### PR TITLE
Avoid pass internalkey to user-defined compaction filter

### PR DESCRIPTION
**Pull request by guoxiangCN**
Fix https://github.com/tikv/titan/issues/285

### CODE REVIEW

File: src/compaction_filter.h [hunk 1]
```diff
@@ -37,20 +37,38 @@ class TitanCompactionFilter final : public CompactionFilter {
 
   Decision FilterV3(int level, const Slice &key, SequenceNumber seqno,
                     ValueType value_type, const Slice &value,
-                    std::string *new_value,
-                    std::string *skip_until) const override {
+                    std::string *new_value, std::string *skip_until) const
+      override {
+    Status s;
+    Slice user_key = key;
+    if (value_type == kBlobIndex) {
```
Comment (Connor1996) [inline src/compaction_filter.h:44]: add a comment here

**Review by Connor1996 — APPROVED** [review]
rest LGTM

**Review by tabokie — APPROVED** [review]

File: src/compaction_filter.h (hunk 1 above)
Comment (guoxiangCN) [reply src/compaction_filter.h:48]: done

**Review by Connor1996 — COMMENTED** [review]
Please add a test case to cover it

**Comment by guoxiangCN** [comment]
done.  the code was formatted by clang-format-diff, but the jenkens-ci always failed in format check task....

**Comment by guoxiangCN** [comment]
Hi, [Connor1996](https://github.com/Connor1996)  could u help merge this pr before this weekend, or else we will miss the release window.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by Connor1996** [comment]
Please fix the format @guoxiangCN

**Comment by guoxiangCN** [comment]
> Please fix the format @guoxiangCN

i had format with
```
clang-format -style=google -i ./src/compaction_filter.h
clang-format -style=google -i ./src/compaction_filter_test.cc
```
but the ci always failed. 
Is the logic of clang-format and clang-format-diff different?
May I ask for the correct formatting method?

**Comment by Connor1996** [comment]
@guoxiangCN still has format issue

**Comment by guoxiangCN** [comment]
@Connor1996 done, thks, It is indeed due to different versions of the clang format.

**Comment by guoxiangCN** [comment]
@Connor1996 maybe this pr should be cherry pick into all, tikv-6.x, or else the gc logic maybe fatal

**Comment by Connor1996** [comment]
@guoxiangCN TiKV's compaction filter is only added on write cf while titan is enabled on default cf. So TiKV isn't affected by this. If you have some needs, you can pick it.

**Pull request merged**